### DISCUSSION TITLE
[Feature] search by stars

### DISCUSSION THREAD

**Discussion by ilyaChuk** [immich-app/immich#11777] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

The function of setting and displaying stars has appeared. It would be much more useful if it was possible to specify the number of stars in a search

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by stumpigit** [comment]
Thank you for your feedback. I will soon create a PR that allows filtering by stars. The functionality is already implemented, I still need to write the tests.

I am confident that this function will be available in the next release.

![starratingfilter](https://github.com/user-attachments/assets/9bd82f06-d099-4185-b43f-e1b5743f06ac)

**Reply by benjaminv2** [reply]
Thank you! it's a important improvement to manage my 100k+ photos :)

**Reply by t-wang-dev** [reply]
I haven't been able to find the settings to enable searching by star ratings, is there one? Or this is still pending to be included in a prod release? Thanks @stumpigit

**Reply by DomDomonom** [reply]
Great! This would be game changing for me as a professional photographer with my archive all rated! Looking forward to it joining prod!

**Reply by Chreutz** [reply]
@stumpigit This is such a tease 🤣. You even posted a screenshot of it working. 

But no worries. Hope you are doing well and I am looking forward to this feature eventually coming 😄

**Reply by stumpigit** [reply]
I apologise for not having worked on it further. I really did have a working version with an associated PR in it. However, this was not adopted and now some things have changed in the backend. I will look into this again over the coming holidays.

**Reply by DomDomonom** [reply]
Don’t apologise! We appreciate the effort! Don’t work too hard over the holidays!

**Reply by Chreutz** [reply]
Apologies for being snarky 🙏. I couldn't find the code anywhere on your profile so I had no idea. Don't stress over it ❤️

**Reply by AgentBignose** [reply]
@stumpigit I am so looking forward to this! :)

**Reply by AgentBignose** [reply]
It's there in 1.127.0 amazing!! thx so much.
I use exif rating a lot and immich just become so much more useful to me!!

Suggestion for next improvement: Have =; >=; <=, >; <; <> logic included.

**Comment by AlessandroLorenzi** [comment]
Hi even if i have version [v1.129.0](https://github.com/immich-app/immich/pkgs/container/immich-server) installed i still no have this feature.

Hi have to enable it somehow? Thanks 🙏🏻 :) 

![image](https://github.com/user-attachments/assets/c8ed0f02-d77d-464f-a74a-d3106bb88646)


![image](https://github.com/user-attachments/assets/309349c2-1131-446e-b576-bd62a9e51376)

**Reply by alextran1502** [reply]
@AlessandroLorenzi hello, you need to go to Account Settings > Feature > Star Rating then enable it

**Reply by AlessandroLorenzi** [reply]
I've never noticed that, thanks a lot 🙏🏻

**Reply by ajayjohn** [reply]
Does this work on the mobile apps as well? I have the latest Android app and I still just see the favorite functionality and not the stars.

**Reply by DomDomonom** [reply]
My understanding is that searching by stars is only in the web interface for now. I'm also waiting for full support.

**Comment by Nick468** [comment]
Hey @bo0tzz could you reopen this feature request as the main part of the feature (filtering by =; >=; <=, >; <; <>) as mentioned [above](https://github.com/immich-app/immich/discussions/11777?sort=old#discussioncomment-12343426) in is not yet implemented? I consider this an essential part of searching for stars as you generally search images above a certain threshold.

**Reply by bo0tzz** [reply]
General search improvements to allow that sort of thing are being considered; while that's in the pipeline we don't really want to add a special case to do it just for one field.

**Reply by anarcat** [reply]
still, i think it might be useful for other users (like me) who have been wondering for a while "how the heck do i search by rating"! :)

if this discussion is kept open, perhaps people will find it easier to understand the ratings are parsed, but can't quite be searched yet (if i understand this right!).

(has someone figured out to map (say) 3+ stars into "favorites" or something yet? maybe as an automated job in the database in the background or a postgresql trigger or something... doesn't seem impossible as a third-party hack..? asking for a friend ;))

thanks for immich anyways, it's pretty shiny and works well!

**Reply by dehnert** [reply]
Is there  an issue that I can follow to track "search by at least N stars"? Even if it's not ready to implement yet, it's weird not to have a tracking ticket. [Edit: #17616]

**Comment by stueble** [comment]
As Nick468 mentioned, it would be much more helpful if it would be possible to search for a certain number of stars and above/below. This could also be implemented by allowing the user to select different ratings in the filter, e.g., to select 3, 4, and 5 stars to show (i.e., with checkboxes). Since this feature request has been closed, should I open another one or is there already a related feature request?

**Comment by Nick468** [comment]
For anyone still annoyed, I fixed my usecase by editing the docker-compose file and modifying the search logic (I am too lazy to make a fork for this). Searching for stars now returns images with >= x stars. This is also useful for ImmichFrame as now you can specify a minimum star rating for images. I just edited the entrypoint for immich-server:
```
    entrypoint:
      - sh
      - -c
      - |
        echo "Patching rating search (dirty)"
        sed -i "s/options.rating === null ? 'is' : '=',/options.rating === null ? 'is' : '>=',/g" server/dist/utils/database.js
        exec start.sh "$@"
```

**Reply by Chreutz** [reply]
Well, that certainly is a solution 😅. Thanks for sharing!
